From 641d34b25c03a761ee525db62eceb8d87a5fd3ac Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 07 Oct 2019 11:37:54 +0300
Subject: [PATCH] [SW-0] hl: turn preboot_sync_btl() per device

It is primarly aim to avoid UART contention between BTL PCIe image and
preboot
In Goya timing is critical since prstn and rstn are tied so we cannot
afford another wait.
In Gaudi prstn and rstn are not tied hence we are less time stressd
(<20mS).
This give us the time to wait till BTL PCIe section is done and see its
log ID (good for debug)

Change-Id: I2f5f3edabcc1ffcb60cf36a9464efac4bcc4c490
---

diff --git a/common/preboot.c b/common/preboot.c
index 9265183..44b3b21 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -144,16 +144,6 @@
 static inline void preboot_version(void) {}
 #endif	/* TARGET_IS_TESTER */
 
-/*
- * Wait for Boot Manager to complete PCIe image execution
- * and return to idle state, so it won't initiate SPI access anymore
- */
-static void preboot_sync_btl(void)
-{
-	while (regmap_readl(mmPSOC_GLOBAL_CONF_BTM_FSM))
-		;
-}
-
 static void hl_jtag_init(void)
 {
 	u32 reg, nPRESETDBG = BIT(12);
diff --git a/common/preboot.h b/common/preboot.h
index e17b16b..488a535 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -57,5 +57,6 @@
 void preboot_pcie_wait_for_link(void);
 void preboot_delay(void);
 void preboot_putc_virt_uart(const char c);
+void preboot_sync_btl(void);
 
 #endif /* COMMON_PREBOOT_H_ */
diff --git a/gaudi/common.c b/gaudi/common.c
index 762c79c..fd6b559 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -70,3 +70,12 @@
 {
 	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
 }
+/*
+ * Wait for Boot Manager to complete PCIe image execution
+ * and return to idle state, so it won't initiate UART access anymore
+ */
+void preboot_sync_btl(void)
+{
+	while (regmap_readl(mmPSOC_GLOBAL_CONF_BTM_FSM))
+		;
+}
diff --git a/goya/common.c b/goya/common.c
index 2b65955..615826d 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -168,3 +168,4 @@
 }
 
 void hl_pll_wa(void) {}
+void preboot_sync_btl(void) {}
